receiver/sapmreceiver/trace_receiver.go

```diff
@@ -44,9 +44,10 @@ var gzipWriterPool = &sync.Pool{
 // sapmReceiver receives spans in the Splunk SAPM format over HTTP
 type sapmReceiver struct {
 	settings component.TelemetrySettings
+	config   *Config
 
-	config *Config
-	server *http.Server
+	server     *http.Server
+	shutdownWG sync.WaitGroup
```

Comment:
Are these changes required for enabling receivers in the tests?

Reply:
Correct, right now the component was incorrect because it did not wait for the server to close before exiting shutdown.

Reply:
Without this the tests fail with binding already bounded port since server is not yet closed.

Reply:
Ok, thanks for clarifying